stellargraph/losses.py

```diff
@@ -19,21 +19,30 @@
 from .core.experimental import experimental
 
 
-@experimental(reason="lack of unit tests", issues=[804])
-def graph_log_likelihood(y_true, y_pred):
+def graph_log_likelihood(batch_adj, wys_output):
     """
+    Computes the graph log likelihood loss function as in https://arxiv.org/abs/1710.09599
+
+    Args:
+        batch_adj: tensor with shape (batch_rows, 1, num_nodes) containing rows of the adjacency matrix
+        wys_output: tensor with shape (batch_rows, 2, num_nodes) containing the embedding outer product
+            scores (with shape (batch_rows, 1, num_nodes) and attentive expected random walk
+            (with shape (batch_rows, 1, num_nodes) concatenated.
+    Returns:
+        the graph log likelihood loss for the batch
     """
-    batch_adj = tf.gather(y_true, [0], axis=1)
 
-    expected_walks = tf.gather(y_pred, [0], axis=1)
-    sigmoids = tf.gather(y_pred, [1], axis=1)
+    batch_adj = tf.squeeze(batch_adj, axis=1)
```

Comment:
Before I was gathering on `axis` of the `(batch_size, 1, num_nodes)` tensor to get a `(batch_size, num_nodes)` tensor.

so in this case its equivalent to the `gather` operation,  just slightly more optimal because as I understand it `squeeze` doesnt change the underlying data structure.